segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/InMemoryChunkStorage.java

```diff
@@ -0,0 +1,295 @@
+        Preconditions.checkNotNull(chunkName);
+        InMemoryChunk chunk = chunks.get(chunkName);
+        if (null == chunk) {
+            throw new ChunkNotFoundException(chunkName, "InMemoryChunkStorage::doGetInfo");
+        }
+        return ChunkInfo.builder()
+                .length(chunk.getLength())
+                .name(chunkName)
+                .build();
+    }
+
+    @Override
+    protected ChunkHandle doCreate(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        Preconditions.checkNotNull(chunkName);
+        if (null != chunks.putIfAbsent(chunkName, new InMemoryChunk(chunkName))) {
+            throw new ChunkAlreadyExistsException(chunkName, "InMemoryChunkStorage::doCreate");
+        }
+        return new ChunkHandle(chunkName, false);
+    }
+
+    @Override
+    protected boolean checkExists(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        return chunks.containsKey(chunkName);
+    }
+
+    @Override
+    protected void doDelete(ChunkHandle handle) throws ChunkStorageException, IllegalArgumentException {
+        InMemoryChunk chunk = getInMemoryChunk(handle);
+        if (null == chunk) {
+            throw new ChunkNotFoundException(handle.getChunkName(), "InMemoryChunkStorage::doDelete");
+        }
+        if (chunk.isReadOnly) {
+            throw new ChunkStorageException(handle.getChunkName(), "chunk is readonly");
+        }
+        chunks.remove(handle.getChunkName());
+    }
+
+    @Override
+    protected ChunkHandle doOpenRead(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        if (chunks.containsKey(chunkName)) {
+            return new ChunkHandle(chunkName, true);
+        }
+        throw new ChunkNotFoundException(chunkName, "InMemoryChunkStorage::doOpenRead");
+    }
+
+    @Override
+    protected ChunkHandle doOpenWrite(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        if (chunks.containsKey(chunkName)) {
+            return new ChunkHandle(chunkName, false);
+        }
+        throw new ChunkNotFoundException(chunkName, "InMemoryChunkStorage::doOpenWrite");
+    }
+
+    @Override
+    protected int doRead(ChunkHandle handle, long fromOffset, int length, byte[] buffer, int bufferOffset) throws ChunkStorageException, NullPointerException, IndexOutOfBoundsException {
+        InMemoryChunk chunk = getInMemoryChunk(handle);
+        if (fromOffset >= chunk.getLength()) {
+            throw new ChunkStorageException(handle.getChunkName(), "Attempt to read at wrong offset");
+        }
+        if (length == 0) {
+            throw new ChunkStorageException(handle.getChunkName(), "Attempt to read 0 bytes");
+        }
+
+        // This is implemented this way to simulate possibility of partial read.
+        InMemoryChunkData matchingData = null;
+
+        // TODO : This is OK for now. This is just test code, but need binary search here.
```

Comment:
Please do not leave TODOs in the code without an associated issue:

https://github.com/pravega/pravega/wiki/Contributing#todos

Reply:
fixed.